### PR TITLE
feat(ironfish): add spends to `wallet/getTransactions`

### PR DESCRIPTION
**Pull request by holahula**
## Summary
Adds spends to `wallet/getTransactions`

Closes IFL-736

## Testing Plan
Updated existing unit tests

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ X] Yes
```
https://github.com/iron-fish/website/pull/400

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/rpc/routes/wallet/getTransactions.test.ts [hunk 1]
```diff
@@ -115,28 +116,50 @@ describe('Route wallet/getAccountTransactions', () => {
...
 
-    const responseTransaction = transactions[0]
-    Assert.isNotUndefined(responseTransaction.notes)
+    Assert.isNotUndefined(minerTxn.notes)
+    Assert.isNotUndefined(minerTxn.spends)
 
-    expect(responseTransaction.notes).toHaveLength(1)
+    expect(minerTxn.spends).toHaveLength(0)
+    expect(minerTxn.notes).toHaveLength(1)
   })
```
Comment (NullSoldier) [inline ironfish/src/rpc/routes/wallet/getTransactions.test.ts:164]: Can we just write a simple new test for this?

File: ironfish/src/rpc/routes/wallet/getTransactions.test.ts [hunk 2]
```diff
@@ -115,28 +116,50 @@ describe('Route wallet/getAccountTransactions', () => {
...
-    expect(transactions).toHaveLength(1)
+    expect(transactions).toHaveLength(2)
+
+    const [spendTxn, minerTxn] = transactions
+
+    Assert.isNotUndefined(spendTxn.notes)
+    Assert.isNotUndefined(spendTxn.spends)
+
+    expect(spendTxn.spends).toHaveLength(transaction.spends.length)
+    expect(spendTxn.spends).toHaveLength(1)
```
Comment (ygao76) [inline ironfish/src/rpc/routes/wallet/getTransactions.test.ts:143]: Can deleted this since it checks the same thing as line 142

**Review by ygao76 — COMMENTED** [review]
looks good, can you add the response spends check in other existing unit tests in getTransaciont.test.ts

eg.
```
 expect(transactions[0].spends).toEqual(block.transactions[0].getSpend(0).map{...})
```

**Review by ygao76 — APPROVED** [review]
Approve with the condition the comment is addressed

File: ironfish/src/rpc/routes/wallet/getTransactions.ts [hunk 3]
```diff
@@ -55,7 +57,8 @@ export const GetAccountTransactionsRequestSchema: yup.ObjectSchema<GetAccountTra
       limit: yup.number().notRequired(),
       offset: yup.number().notRequired(),
       confirmations: yup.number().notRequired(),
-      notes: yup.boolean().notRequired(),
+      notes: yup.boolean().notRequired().default(false),
+      spends: yup.boolean().notRequired().default(false),
```
Comment (ygao76) [inline ironfish/src/rpc/routes/wallet/getTransactions.ts:61]: this needs to update to spends schema

**Pull request merged**